### DISCUSSION TITLE
frontmatter is not found on the collection item

### DISCUSSION THREAD

**Discussion by roma-glushko** [withastro/docs#3684] [Ideas]
I'm basically trying to add a readingTime value for my blog posts following suggestion from the receipt:
https://docs.astro.build/en/recipes/reading-time/

The new remark plugin was registered:
<img width="664" alt="Screenshot 2023-07-09 at 20 41 16" src="https://github.com/withastro/docs/assets/9402690/3de8b25a-8803-411b-a416-b0343daa4760">

<img width="707" alt="Screenshot 2023-07-09 at 20 40 25" src="https://github.com/withastro/docs/assets/9402690/14492489-084a-45f7-8f0b-473ef90557b4">

It's being called when I head to the item view page (verified by console.log() in the plugin).
However, I could retrieve the value to render it on the page. My item element doesn't contain any frontmatter fields:

```json
{
  id: 'slug.mdx',
  slug: 'slug',
  body: 'A lot of text...',
  collection: 'blog',
  data: {
    id: 'abc',
    title: 'Title,
    excerpt: 'Descr',
    keywords: [ 'tag1', 'tag2' ],
    pubDate: 2022-06-06T21:00:00.000Z,
    cover: {
      image: '/blog/slug/hero.jpeg'
    }
  },
  render: [AsyncFunction: render]
}

```

I'm curios how should I pull that `readingTimeMIn` value from 🤔

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

Not sure how it worked in 2023, but now the solution is to use `const { remarkPluginFrontmatter } = await render(entry);` as documented in the recipe. So, I'm closing this as resolved.

If you think this is a mistake and that this discussion should remain open, please feel free to ping me, I'll take a second look at it.

Thank your for your contribution!
